File: problems/9-palindrome-number/javascript/solution-2.js
/**
 * I personally love this algorithm, we made up new value each digit by multiply each digit by 10.
 * Then compare it later to see if they have the same value.
 * @param {number} x
 * @return {boolean}
 */
 var isPalindrome = function(x) {
    if(x < 0)
        return false;
    
    let ret = 0;
    for(let i = x; i >= 1; i = Math.floor(i / 10))
        ret = ret * 10 + i % 10;
            
    return ret === x;
};

File: problems/9-palindrome-number/python/solution-2.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x % 10 == 0 and x != 0):
            return False

        reverse_half = 0

        while x > reverse_half:
            reverse_half = reverse_half * 10 + x % 10
            x //= 10

        return x == reverse_half or (x == reverse_half // 10)